sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -307,6 +310,27 @@ export class ManagementClient extends LinkEntity {
 
       const waitTimer = setTimeout(actionAfterTimeout, retryTimeoutInMs);
 
+      const rejectOnAbort = () => {
+        const requestName = sendRequestOptions.requestName;
+        const desc: string =
+          `[${this._context.namespace.connectionId}] The request "${requestName}" ` +
+          `to has been cancelled by the user.`;
+        log.error(desc);
+        const error = new AbortError(
+          `The ${requestName ? requestName + " " : ""}operation has been cancelled by the user.`
+        );
+
+        throw error;
+      };
+
+      if (sendRequestOptions.abortSignal) {
+        if (sendRequestOptions.abortSignal.aborted) {
+          return rejectOnAbort();
+        }
+        // TODO: init() should respect the abort signal as well.
+        // See https://github.com/Azure/azure-sdk-for-js/issues/4422
+      }
```

Comment:
It might be better to move this above [const waitTimer = setTimeout(actionAfterTimeout, retryTimeoutInMs);](https://github.com/Azure/azure-sdk-for-js/pull/8159/files#diff-9faa4c27d90e5b6c44af0553384fb39cR311). That way if the request is already aborted we don't call `setTimeout` at all.

Reply:
Fixed in https://github.com/Azure/azure-sdk-for-js/pull/8159/commits/c1a0e44e43bdb7115820c7fafca30556183ba4f1